Lection 3/task 3.1/ReportApp/Program.cs

```diff
@@ -1,4 +1,23 @@
-﻿using ReportApp.Services;
+﻿using ReportApp.Models;
+using ReportApp.Models.Activity;
+using ReportApp.Models.Shop;
+using ReportApp.Services;
 
-var service = new ActivityReportGeneratorService();
-service.GenerateReport();
\ No newline at end of file
+
+var request = new ConfigurationService();
+var data = request.LoadConfiguration<RequestModel>("./Request/Request.json");
+
+switch (data.Type)
+{
+    case "Activity":
+        var activityService = new BaseReportGeneratorService();
+        activityService.GenerateReport<ActivityReportConfiguration, ActivityReportModel>(data.PathToFile, data.Type);
```

Comment:
Hm, looks ok, but why don't we just creating ActivityReportGeneratorService, which inherits from BaseReportGeneratorService? We can get rid of generics if so